### ISSUE TITLE
SourceMap issue: `sourceMapName` option is somewhat misleading

### ISSUE THREAD

**Issue by getify** [babel/babel#1568]
The `sourceMapName` option says to me (and I imagine others) "the name of the source map". In other words, if you're using `sourceMaps: true` to produce a source map that you intend to write out to a file like `foo.js.map`, you might very well assume (like I have for quite awhile) that you should set `sourceMapName` to `"foo.js.map"`.

It's a very subtle point, but this option is **not** supposed to include the (file)name of the source map. It sets the `file` property in the map, which according to [the SourceMaps spec v3](https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit#heading=h.hval2vfopveu), is actually supposed to be the **target of the sourcemap**.

In other words, you should set `sourceMapName` to `"foo.js"`, not `"foo.js.map"`.

I will be filing another issue/PR with the site repo to improve the docs around this topic, but here I am suggesting this option name itself should change to reduce the default confusions I think are likely (and I have observed with others using babel besides myself).

Instead of `sourceMapName`, I would suggest it change to be `sourceMapTarget`.

**Comment by sebmck** [comment]
Good idea. Unsure why it's named `sourceMapName` as yeah, it's completely counter intuitive. Might have copied it from somewhere.

**Comment by getify** [comment]
:+1:

**Comment by thejameskyle** [comment]
> Might have copied it from somewhere.

https://github.com/benjamn/recast#source-maps ?

**Comment by sebmck** [comment]
@thejameskyle Aha, that'd be it.

**Comment by 3y3** [comment]
What you think about files generated by babel?
They have same name as their sources. Is it correct?

**Comment by getify** [comment]
:+1:
